kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -463,4 +463,94 @@ class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
+
+  test("getSnapshotAtVersion: basic end-to-end read") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+      (0 to 10).foreach { i =>
+        spark.range(i*10, i*10 + 10).write
+          .format("delta")
+          .mode("append")
+          .save(path)
+      }
+      // Read a checkpoint version
+      checkTable(
+        path = path,
+        expectedAnswer = (0L to 99L).map(TestRow(_)),
+        version = Some(9),
+        expectedVersion = Some(9)
+      )
+      // Read a JSON version
+      checkTable(
+        path = path,
+        expectedAnswer = (0L to 89L).map(TestRow(_)),
+        version = Some(8),
+        expectedVersion = Some(8)
+      )
+      // Read the current version
+      checkTable(
+        path = path,
+        expectedAnswer = (0L to 109L).map(TestRow(_)),
+        version = Some(10),
+        expectedVersion = Some(10)
+      )
+      // Cannot read a version that does not exist
+      val e = intercept[RuntimeException] {
+        Table.forPath(defaultTableClient, path)
+          .getSnapshotAtVersion(defaultTableClient, 11)
+      }
+      assert(e.getMessage.contains(
+        "Trying to load a non-existent version 11. The latest version available is 10"))
+    }
+  }
+
+  test("getSnapshotAtVersion: end-to-end test with truncated delta log") {
+    withTempDir { tempDir =>
+      val tablePath = tempDir.getCanonicalPath
+      // Write versions [0, 10] (inclusive) including a checkpoint
+      (0 to 10).foreach { i =>
+        spark.range(i*10, i*10 + 10).write
+          .format("delta")
+          .mode("append")
+          .save(tablePath)
+      }
+      val log = org.apache.spark.sql.delta.DeltaLog.forTable(
+        spark, new org.apache.hadoop.fs.Path(tablePath))
+      // Delete the log files for versions 0-9, truncating the table history to version 10
+      (0 to 9).foreach { i =>
+        val jsonFile = org.apache.spark.sql.delta.util.FileNames.deltaFile(log.logPath, i)
+        new File(new org.apache.hadoop.fs.Path(log.logPath, jsonFile).toUri).delete()
+      }
+      // Create version 11 that overwrites the whole table
+      spark.range(50).write
+        .format("delta")
+        .mode("overwrite")
+        .save(tablePath)
+
+      // Cannot read a version that has been truncated
+      val e = intercept[RuntimeException] {
+        Table.forPath(defaultTableClient, tablePath)
+          .getSnapshotAtVersion(defaultTableClient, 9)
+      }
+      assert(e.getMessage.contains("Unable to reconstruct state at version 9"))
+      // Can read version 10
+      checkTable(
+        path = tablePath,
+        expectedAnswer = (0L to 109L).map(TestRow(_)),
+        version = Some(10),
+        expectedVersion = Some(10)
+      )
+      // Can read version 11
+      checkTable(
+        path = tablePath,
+        expectedAnswer = (0L until 50L).map(TestRow(_)),
+        version = Some(11),
+        expectedVersion = Some(11)
+      )
+    }
+  }
```

Comment:
Do we have a coverage where one commit file in the middle is missing. for example: 10.checkpont, 11.log, <missing>, 13.log.

Reply:
I think this would go in `SnapshotManagerSuite` but it looks like we don't for that specific case. Adding one.